### ISSUE TITLE
Rule Change: Allow comments for banned words

### ISSUE THREAD

**Issue by IAfanasov** [eslint/eslint#16587]
### What rule do you want to change?

id-denylist

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
const StyledDiv = styled.div`
   height: 100vh;
`
```

### What does the rule currently do for this code?

When `100vh` is added to deny list, rule warns that `100vh` is banned.

### What will the rule do after it's changed?

I would like to be able to add reasoning and hints for resolutions. So that message would be something like "Avoid using 100vh because mobile devices consider address bar as part of the 100vh and the screen looks cropped in the bottom or top. Prefer 100%". 
The message should be configurable for each banned word. Might be nice to configure severity level per each word as well. 

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @IAfanasov, thanks for the issue!

I'm not sure if this is related to the ESLint core rule `id-denylist` as it restricts JS identifiers (e.g., variable names), not words in literals, so it wouldn't report `100vh` in the above example. Here's a [playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IGlkLWRlbnlsaXN0OiBbXCJlcnJvclwiLCBcIjEwMHZoXCJdICovXG5cbmNvbnN0IFN0eWxlZERpdiA9IHN0eWxlZC5kaXZgXG4gICBoZWlnaHQ6IDEwMHZoO1xuYCIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnsiZXM2Ijp0cnVlfX19). Either way, per our rules policies [stylistic rules are frozen](https://github.com/eslint/eslint#stylistic-rule-updates) so we are not adding new configuration options to `id-denylist`. You can always copy the existing rule into a custom rule and add new options.
